IdentityCore/src/cache/mac/MSIDMacCredentialStorageItem.m

```diff
@@ -158,57 +176,41 @@ - (NSDictionary *)jsonDictionary
     return dictionary;
 }
 
-- (MSIDDefaultCredentialCacheKey *)getKeyForCredential:(MSIDCredentialCacheItem *)credential
+- (MSIDCacheKey *)createKeyFromCredential:(MSIDCredentialCacheItem *)credential
 {
-    MSIDDefaultCredentialCacheKey *credentialKey = [[MSIDDefaultCredentialCacheKey alloc] initWithHomeAccountId:credential.homeAccountId environment:credential.environment clientId:credential.clientId credentialType:credential.credentialType];
+    MSIDDefaultCredentialCacheKey *credentialCacheKey = [[MSIDDefaultCredentialCacheKey alloc] initWithHomeAccountId:credential.homeAccountId environment:credential.environment clientId:credential.clientId credentialType:credential.credentialType];
     
-    credentialKey.familyId = credential.familyId;
-    credentialKey.realm = credential.realm;
-    credentialKey.target = credential.target;
-    credentialKey.enrollmentId = credential.enrollmentId;
-    return credentialKey;
+    credentialCacheKey.familyId = credential.familyId;
+    credentialCacheKey.realm = credential.realm;
+    credentialCacheKey.target = credential.target;
+    credentialCacheKey.enrollmentId = credential.enrollmentId;
+    
+    return [self createKeyFromCredentialCacheKey:credentialCacheKey];
 }
 
-- (NSString *)getCredentialKey:(MSIDDefaultCredentialCacheKey *)key
+- (MSIDCacheKey *)createKeyFromCredentialCacheKey:(MSIDCacheKey *)key
+{
+    MSIDCacheKey *storedCredentialKey = [[MSIDCacheKey alloc] initWithAccount:key.account service:key.service generic:key.generic type:key.type];
```

Comment:
I did some testing with NSPredicate and calculated properties and I didn't see a need to create new cache key from passed in key. NSPredicate should work on calculated properties based on my testing.

Can we double check on why this extra cache key creation is necessary? Perhaps there's something else that's causing unexpected behavior and causes mismatches?

Here's the sample code I've run for reference:

```
// TestFilteredObject.h file
@interface TestFilteredObject : NSObject

@property NSUInteger testNumber;

@property (readonly) NSString *testNumberString;
@property (readonly) NSString *testNumberString2;

@end

// TestFilteredObject.m file
#import "TestFilteredObject.h"

@implementation TestFilteredObject

- (NSString *)testNumberString
{
    return [NSString stringWithFormat:@"%ld", (long)self.testNumber];
}

- (NSString *)testNumberString2
{
    return [NSString stringWithFormat:@"%ld", (long)self.testNumber*2];
}

- (NSString *)description
{
    return [NSString stringWithFormat:@"%@, and %@", self.testNumberString, self.testNumberString2];
}

@end

// Test code

TestFilteredObject *object1 = [TestFilteredObject new];
    object1.testNumber = 1;
    
    TestFilteredObject *object2 = [TestFilteredObject new];
    object2.testNumber = 2;
    
    TestFilteredObject *object3 = [TestFilteredObject new];
    object3.testNumber = 3;
    
    NSArray *nonFilteredArray = @[object1, object2, object3];
    
    NSLog(@"Non filtered array: %@", nonFilteredArray);
    
    NSString *searchTerm = @"3";
    NSString *searchTerm2 = @"6";
    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"testNumberString == %@ AND testNumberString2 == %@", searchTerm, searchTerm2];
    NSArray *filteredArray = [nonFilteredArray filteredArrayUsingPredicate:predicate];
    
    NSLog(@"Filtered array: %@", filteredArray);

// Output
/*
2019-07-07 21:52:18.559754-0700 TestPredicate[60007:12810224] Non filtered array: (
    "1, and 2",
    "2, and 4",
    "3, and 6"
)
2019-07-07 21:52:18.560187-0700 TestPredicate[60007:12810224] Filtered array: (
    "3, and 6"
)
*/

```

Reply:
Hey good catch. I confirmed that nspredicate works with properties. The previous code was not working before because of the way we were copying MSIDCacheKey objects. Updated the code to implement NSCopying on MSIDCredentialCacheKey instead and removed the need to create MSIDCacheKey from the code.